sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -110,16 +113,17 @@ public void pointTo(Object baseObject, long baseOffset, int sizeInBytes) {
     this.baseObject = baseObject;
     this.baseOffset = baseOffset;
     this.sizeInBytes = sizeInBytes;
+    this.headerInBytes = calculateHeaderPortionInBytes(numElements);
```

Comment:
ah i see, it's different from unsafe row. The `numElements` can be different everytime we call `pointTo`